From be91abe7fc8cc731b57bec4cf2c004c07b0fd79b Mon Sep 17 00:00:00 2001
From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Wed, 25 Apr 2018 10:56:16 +0200
Subject: Ignore "migration" key for determining whether the config is empty

Otherwise we would never generate the default power management configuration
leading to it not suspending on lid close, not handling power buttons etc etc

CHANGELOG: Fixed bug that caused power management system to not work on a fresh install

Reviewed-By: David Edmundson

BUG: 391782
FIXED-IN: 5.12.5
---
 daemon/powerdevilcore.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/daemon/powerdevilcore.cpp b/daemon/powerdevilcore.cpp
index 2cf936e..53b7521 100644
--- a/daemon/powerdevilcore.cpp
+++ b/daemon/powerdevilcore.cpp
@@ -108,9 +108,15 @@ void Core::onBackendReady()
 
     m_profilesConfig = KSharedConfig::openConfig("powermanagementprofilesrc", KConfig::CascadeConfig);
 
+    QStringList groups = m_profilesConfig->groupList();
+    // the "migration" key is for shortcuts migration in added by migratePre512KeyboardShortcuts
+    // and as such our configuration would never be considered empty, ignore it!
+    groups.removeOne(QStringLiteral("migration"));
+
     // Is it brand new?
-    if (m_profilesConfig->groupList().isEmpty()) {
+    if (groups.isEmpty()) {
         // Generate defaults
+        qCDebug(POWERDEVIL) << "Generating a default configuration";
         bool toRam = m_backend->supportedSuspendMethods() & PowerDevil::BackendInterface::ToRam;
         bool toDisk = m_backend->supportedSuspendMethods() & PowerDevil::BackendInterface::ToDisk;
         ProfileGenerator::generateProfiles(toRam, toDisk);
-- 
cgit v0.11.2

